### PR TITLE
Add BDPluginSpec constructor and use it in plugin_specs_from_names

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Starting with Python 3.12 we got some weird complains when setting attributes of BlockDev.PluginSpec in plugin_specs_from_names, it's better to add a fully working constructor to create the struct correctly from the Python overrides.

### CODE REVIEW

File: src/lib/plugins.c [hunk 1]
```diff
@@ -40,6 +40,21 @@ void bd_plugin_spec_free (BDPluginSpec *spec) {
...
+ * bd_plugin_spec_new: (constructor)
+ * @name: %BDPlugin name, e.g. %BD_PLUGIN_LVM
+ * @so_name: (nullable): SO name of the plugin to load or %NULL for default
+ *
+ * Returns: (transfer full): a new plugin spec
+ */
+BDPluginSpec* bd_plugin_spec_new (BDPlugin name, const gchar *so_name) {
+    BDPluginSpec* ret = g_new0 (BDPluginSpec, 1);
+    ret->name = name;
+    ret->so_name = so_name ? g_strdup (so_name) : NULL;
```
Comment (tbzatek) [inline src/lib/plugins.c:53]: allocating a const? typecasting it for `g_free()`?

File: src/lib/plugins.c [hunk 2]
```diff
@@ -40,6 +40,21 @@ void bd_plugin_spec_free (BDPluginSpec *spec) {
     g_free (spec);
 }
 
+/**
+ * bd_plugin_spec_new: (constructor)
+ * @name: %BDPlugin name, e.g. %BD_PLUGIN_LVM
+ * @so_name: (nullable): SO name of the plugin to load or %NULL for default
```
Comment (tbzatek) [inline src/lib/plugins.c:46]: `BDPlugin` is `enum`, i.e. `int`.

File: src/lib/plugins.c (hunk 1 above)
Comment (vojtechtrefny) [reply src/lib/plugins.c:53]: I know it's weird, but we already allocate it in `bd_plugin_spec_copy` and free it in `bd_plugin_spec_free` and I didn't want to change it.

Comment (tbzatek) [reply src/lib/plugins.c:53]: Hmm fair enough. Perhaps you could add some `/* FIXME */`s? :-)

**Review by tbzatek — APPROVED** [review]

**Pull request merged**